src/View.js

```diff
@@ -0,0 +1,294 @@
+import { EVENT_TYPES } from './EventEmitter';
+
+const dropPermissionMap = {
+  ore: ['trash', 'inventory', 'craft-slots', 'create-new-recipe'],
+  recipe: ['trash', 'recipe-template'],
+  item: ['trash', 'inventory', 'craft-slots', 'create-new-recipe'],
+};
+
+export default class View {
+  constructor(emitter) {
+    this.emitter = emitter;
+
+    this.inventoryContainer = document.getElementById('inventory');
+    this.recipeSlots = document.querySelector('.recipe');
+    this.newRecipeContainer = document.getElementById('new-recipe');
+    this.craftingContainer = document.getElementById('craft-slots');
+    this.newRecipeSlots = document.getElementById('create-new-recipe');
+    this.recipeContainer = document.getElementById('recipe');
+    this.recipeTemplate = document.getElementById('recipe-template');
+
+    this.slotContainers = {
+      inventory: this.inventoryContainer,
+      recipe: this.recipeSlots,
+      forge: this.craftingContainer,
+      newRecipe: this.newRecipeSlots,
+      trash: this.trashZone,
+      recipeTemplate: this.recipeTemplate,
+    };
+
+    this.emitter.on(EVENT_TYPES.SLOTS_CREATED, slots => this.displaySlots(slots));
+
+    this.mineButton = document.getElementById('mine-button');
+    this.inventorySlotDiv = document.querySelectorAll('.ore');
+    this.infoButton = document.getElementById('display-button');
+    this.closeButton = document.getElementById('close-button');
+    this.modal = document.getElementById('ruleset');
+    this.addNewRecipeButton = document.getElementById('add-recipe');
+    this.cancelButton = document.getElementById('cancel');
+    this.createNewRecipeButton = document.getElementById('create');
+    this.inventorySlot = document.querySelector('.inventory');
+    this.draggableOre = document.getElementsByTagName('button');
+    this.trashZone = document.getElementById('trash');
+    this.forgeButton = document.getElementById('forge');
+    this.recipeNameElement = document.getElementById('recipe-name');
+
+    this.infoButton.addEventListener('click', () => {
+      this.modal.style.display = 'block';
+    });
+    this.closeButton.addEventListener('click', () => {
+      this.modal.style.display = 'none';
+    });
+    this.addNewRecipeButton.addEventListener('click', () => {
+      this.newRecipeContainer.style.display = 'block';
+      this.recipeContainer.replaceWith(this.newRecipeContainer);
+      this.emitter.emit(EVENT_TYPES.CLICK_ADD);
+    });
+    this.cancelButton.addEventListener('click', () => {
+      this.emitter.emit(EVENT_TYPES.CLICK_CANCEL);
+      this.newRecipeContainer.replaceWith(this.recipeContainer);
+    });
+    this.createNewRecipeButton.addEventListener('click', () => this.saveNewRecipeInfo());
+
+    this.mineButton.addEventListener('click', () => {
+      this.emitter.emit(EVENT_TYPES.CLICK_MINE);
+    });
+    this.forgeButton.addEventListener('click', () => {
```

Comment:
can be "one liner" if there is only one expression in arrow function - {} can be omitted